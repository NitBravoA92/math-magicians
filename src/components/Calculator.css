#calculator-app {
  width: 450px;
  border: none;
  position: relative;
  background-color: transparent;
}

.display-result {
  background-color: gray;
  padding: 0.8rem 0.4rem;
  width: 100%;
  min-height: 60px;
  font-size: 18px;
  display: flex;
  flex-direction: column;
  gap: 0.6rem;
  text-align: right;
  color: #fff;
}

.previews-result {
  font-size: 0.8em;
}

.final-result {
  font-size: 1.2em;
}
