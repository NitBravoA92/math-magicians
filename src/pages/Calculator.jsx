import Layout from './Layout';
import CalculatorApp from '../components/CalculatorApp';

const Calculator = () => (
  <Layout>
    <section id="calculator" className="d-column">
      <h1 className="page-title">Let&apos;s do some math!</h1>
      <div className="page-content">
        <CalculatorApp />
      </div>
    </section>
  </Layout>
);

export default Calculator;
