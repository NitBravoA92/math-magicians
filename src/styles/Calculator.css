#calculator-app {
  width: 450px;
  border: none;
  position: relative;
  background-color: transparent;
  font-size: 18px;
}

.display-result {
  background-color: gray;
  padding: 0.8rem 0.4rem;
  width: 100%;
  min-height: 60px;
  display: flex;
  align-items: center;
  text-align: right;
  color: #fff;
}

.final-result {
  width: 100%;
  font-size: 1.3em;
}

.buttons-container {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
}

.btn {
  border: none;
  outline: none;
  cursor: pointer;
  padding: 1.3rem 1rem;
  border-bottom: 1px solid #d4d4d4;
  font-size: 0.9em;
}

.btn-gray {
  border-right: 1px solid #d4d4d4;
  background-color: #eaeaea;
}

.btn-orange {
  background-color: rgb(236, 137, 79);
}

.btn-gray:hover {
  background-color: #d4d4d4;
}

.btn-orange:hover {
  background-color: rgb(235, 121, 56);
}

.btn-zero {
  grid-column: span 2;
}
