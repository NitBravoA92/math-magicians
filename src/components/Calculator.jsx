import './Calculator.css';

function Calculator() {

  return(
    <section id="calculator-app">
      <div className="display-result">
        <p className="previews-result"></p>
        <p className="final-result">0</p>
      </div>
      
    </section>
  );
}

export default Calculator;