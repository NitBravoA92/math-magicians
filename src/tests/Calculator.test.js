import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Calculator from '../pages/Calculator';

describe('The Calculator component', () => {
  test('renders correctly into the DOM', () => {
    const component = renderer
      .create(
        <MemoryRouter>
          <Calculator />
        </MemoryRouter>,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  test("should render the text: Let's do some math!", () => {
    const component = render(
      <MemoryRouter>
        <Calculator />
      </MemoryRouter>,
    );

    expect(component.container).toHaveTextContent("Let's do some math!");
  });

  test("should render the CalculatorApp with 19 buttons and displays the value '0' as default", () => {
    const component = render(
      <MemoryRouter>
        <Calculator />
      </MemoryRouter>,
    );

    const calculatorContainer = component.container.querySelector('#calculator-app');
    const calculatorButtons = component.container.querySelectorAll('button');
    const displayResult = component.container.querySelector('.final-result');

    expect(calculatorContainer).toBeInTheDocument();
    expect(calculatorButtons.length).toBe(19);
    expect(displayResult.textContent).toBe('  0 ');
  });
});
