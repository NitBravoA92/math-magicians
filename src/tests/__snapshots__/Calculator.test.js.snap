// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Calculator component renders correctly into the DOM 1`] = `
Array [
  <header>
    <a
      className="logo-container"
      href="/"
    >
      <img
        alt="logo"
        className="App-logo"
        src="mathLogo.gif"
      />
      <span
        className="logo-title"
      >
        Math Magicians
      </span>
    </a>
    <nav>
      <ul
        id="menu"
      >
        <li
          className="nav-item"
        >
          <a
            href="/"
          >
            Home
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            href="/calculator"
          >
            Calculator
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            href="/quote"
          >
            Quote
          </a>
        </li>
      </ul>
    </nav>
  </header>,
  <main>
    <section
      id="calculator"
    >
      <h1
        className="page-title"
      >
        Let's do some math!
      </h1>
      <div
        className="page-content"
      >
        <section
          id="calculator-app"
        >
          <div
            className="display-result"
          >
            <p
              className="final-result"
            >
                0 
            </p>
          </div>
          <div
            className="buttons-container"
          >
            <button
              className="btn btn-gray"
              onClick={[Function]}
              type="button"
            >
              AC
            </button>
            <button
              className="btn btn-gray"
              onClick={[Function]}
              type="button"
            >
              +/-
            </button>
            <button
              className="btn btn-gray"
              onClick={[Function]}
              type="button"
            >
              %
            </button>
            <button
              className="btn btn-orange"
              onClick={[Function]}
              type="button"
            >
              รท
            </button>
            <button
              className="btn btn-gray"
              onClick={[Function]}
              type="button"
            >
              7
            </button>
            <button
              className="btn btn-gray"
              onClick={[Function]}
              type="button"
            >
              8
            </button>
            <button
              className="btn btn-gray"
              onClick={[Function]}
              type="button"
            >
              9
            </button>
            <button
              className="btn btn-orange"
              onClick={[Function]}
              type="button"
            >
              x
            </button>
            <button
              className="btn btn-gray"
              onClick={[Function]}
              type="button"
            >
              4
            </button>
            <button
              className="btn btn-gray"
              onClick={[Function]}
              type="button"
            >
              5
            </button>
            <button
              className="btn btn-gray"
              onClick={[Function]}
              type="button"
            >
              6
            </button>
            <button
              className="btn btn-orange"
              onClick={[Function]}
              type="button"
            >
              -
            </button>
            <button
              className="btn btn-gray"
              onClick={[Function]}
              type="button"
            >
              1
            </button>
            <button
              className="btn btn-gray"
              onClick={[Function]}
              type="button"
            >
              2
            </button>
            <button
              className="btn btn-gray"
              onClick={[Function]}
              type="button"
            >
              3
            </button>
            <button
              className="btn btn-orange"
              onClick={[Function]}
              type="button"
            >
              +
            </button>
            <button
              className="btn btn-gray btn-zero"
              onClick={[Function]}
              type="button"
            >
              0
            </button>
            <button
              className="btn btn-gray"
              onClick={[Function]}
              type="button"
            >
              .
            </button>
            <button
              className="btn btn-orange"
              onClick={[Function]}
              type="button"
            >
              =
            </button>
          </div>
        </section>
      </div>
    </section>
  </main>,
]
`;
